javascript/ql/src/semmle/javascript/dataflow/Configuration.qll

```diff
@@ -298,6 +310,15 @@ abstract class BarrierGuardNode extends DataFlow::Node {
    */
   abstract predicate blocks(boolean outcome, Expr e);
 
+  /**
+   * Holds if this barrier guard blocks all labels.
+   */
+  predicate blocksAllLabels() { any() }
+
+  /**
+   * Holds if this barrier guard only blocks specific labels, and `label` is one of them.
+   */
+  predicate blocksSpecificLabel(FlowLabel label) { none() }
```

Comment:
`blocksAllLabels` and `blocksSpecificLabel` are implied by each other, do we really need both? 
We should at least document that none or both should be overridden.

Reply:
I've added documentation about the overrides.

I agree it's a bit ugly to have both, but defaulting to `any()` is expensive. Inefficient APIs tend to get hacks and workarounds built on top of them, which I think is the greater evil.

Reply:
`blocks(FlowLabel label)` clashes with `blocks(DataFlow::Node node)` as QL doesn't have type-based overloading.

`isSpecificTo(FlowLabel label)`?

Reply:
That's a shame. I note, though, that `blocks(DataFlow::Node)` is marked internal, so we can rename it.

Reply:
I'm not convinced about `blocks`. Can we please agree on another name?

If the `TaintKind` alias means we can't name predicates properly then we should remove the alias instead of being indecisive about it.

Reply:
Sure, I'm not particularly attached to calling it `blocks`, I just want to keep the name simple. You seem to have much stronger opinions on this than me, so I'll just leave it up to you.